### ISSUE TITLE
update the tests for twisted.web.static to approach current standards

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3549]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3549|
|Type|task|
|Created|2008-11-17 23:52:06Z|
|Branch|https://github.com/twisted/twisted/tree/web-static-tests-3549|

I started to work on [#3469](https://github.com/twisted/twisted/issues/3469) and I couldn't get anywhere because of the state of the existing tests for `twisted.web.static`.  So I'll improve those tests first.

<details><summary>Searchable metadata</summary>

```
trac-id__3549 3549
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_web_static_tests_3549 branches/web-static-tests-3549
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1226965926000000 1226965926000000
changetime__1227112167000000 1227112167000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [25408]) Branching to 'web-static-tests-3549'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Okay, those tests are a bunch better now.

I thought either this or [[#3469](https://github.com/twisted/twisted/issues/3469)](https://github.com/twisted/twisted/issues/3469) was going to be bigger, but both ended up being small, so I ended up making the #3469 fix in this branch as well.  It just consisted of a couple new tests for 404 behavior and a couple new lines in static.py.  Hopefully it's just as easy to review in this form as it would have been in two different branches.

I expect this branch will resolve both tickets.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

The docstring of `test_securityViolationNotFound` looks weird (the which is not at the correct position). Once fixed please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [25444]) fix the docstring

refs [#3549](https://github.com/twisted/twisted/issues/3549)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [25445]) Merge Twisted web-static-tests-3549

Author: exarkun
Reviewer: therve
Fixes: [#3549](https://github.com/twisted/twisted/issues/3549)
Fixes: [#3469](https://github.com/twisted/twisted/issues/3469)

Add a check to `twisted.web.static.File` for security violations and turn them into 404
responses instead of letting the exception go unhandled and turn into a 500 response.

Also grab the tests for `File` from `test_web` and put them into `test_static` in a
slightly improved form.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
